libmachine/provision/boot2docker.go

```diff
@@ -53,174 +45,62 @@ func (provisioner *Boot2DockerProvisioner) upgradeIso() error {
 
 	if err := provisioner.Driver.Stop(); err != nil {
 		return err
+
 	}
 
 	if err := utils.WaitFor(drivers.MachineInState(provisioner.Driver, state.Stopped)); err != nil {
 		return err
+
 	}
 
 	machineName := provisioner.GetDriver().GetMachineName()
 
 	log.Infof("Upgrading machine %s...", machineName)
 
-	b2dutils := utils.NewB2dUtils("", "")
+	isoFilename := ""
+	switch provisioner.GetDriver().DriverName() {
+	case "virtualbox":
+		isoFilename = "boot2docker-virtualbox.iso"
+	case "vmwarefusion", "vmwarevsphere":
+		isoFilename = "boot2docker-vmware.iso"
+	case "hyperv":
```

Comment:
It's actually `hyper-v` (with a hyphen).  We probably should have constants for the driver names for this reason, I think, but we can file a separate issue for that.